lib/ecto_ldap/adapter.ex

```diff
@@ -37,27 +38,48 @@ defmodule Ecto.Ldap.Adapter do
   # GenServer server API
   #
   ####
-  def handle_call({:search, search_options}, _from, state) do
+  def handle_call({:search, search_options, params}, _from, state) do
+    {:ok, handle}   = ldap_connect(state)
+    search_response = ldap_search(handle, search_options, params)
+    ldap_disconnect(handle)
+
+    {:reply, search_response, state}
+  end
+
+  def handle_call(:base, _from, state) do
+    base = Keyword.get(state, :base) |> to_char_list
+    {:reply, base, state}
+  end
 
+  def ldap_connect(state) do
     user_dn   = Keyword.get(state, :user_dn)  |> to_char_list
     password  = Keyword.get(state, :password) |> to_char_list
-    host      = Keyword.get(state, :host)   |> to_char_list
+    host      = Keyword.get(state, :server)   |> to_char_list
```

Comment:
I think I'd prefer this to be `:hostname`. That's consistent Ecto format w/ the Postgres adapter.